nova/core/roles/os_configuration/tasks/main.yml

```diff
@@ -4,56 +4,64 @@
     - customization_context == "host" # Alternative is container and then it doesn't need os_configuration
     - infra_env != "my_host" # my_host is computer where you run the play from
   block:
-    - name: Including Windows configuration tasks...
-      ansible.builtin.include_tasks: windows.yml
-      when: "'os_windows' in group_names"
-
-    - name: Including Windows in Azure configuration tasks...
-      ansible.builtin.include_tasks: windows_azure.yml
+    - name: Configuring non-Netowork OS ...
       when:
-        - "'os_windows' in group_names"
-        - infra_env == 'azure'
+        - ansible_network_os is not defined
+        - "'os_macos' not in group_names"
+      block:
+        - name: Including Windows configuration tasks...
+          ansible.builtin.include_tasks: windows.yml
+          when: ansible_system == "Win32NT"
+
+        - name: Including Windows on vSphere configuration tasks...
+          ansible.builtin.include_tasks: windows_vsphere.yml
+          when:
+            - ansible_system == "Win32NT"
+            - infra_env in ["vsphere" or "vmware_workstation"]
+
+        - name: Including Linux configuration tasks...
+          ansible.builtin.include_tasks: linux.yml
+          when: ansible_system == "Linux"
+
+        - name: Including Linux in Cloud configuration tasks...
+          ansible.builtin.include_tasks: linux_cloud.yml
+          when:
+            - ansible_system == "Linux"
+            - infra_env in ["aws", "linode", "azure"]
+
+        - name: Including Windows in Cloud configuration tasks...
+          ansible.builtin.include_tasks: windows_cloud.yml
+          when:
+            - ansible_system == "Win32NT"
+            - infra_env in ["aws", "linode", "azure"]
```

Comment:
Could it be recursive files?
Example: `ansible.include_tasks: windows_vsphere.yml` in `windows.yml` file. And same for Linux and windows_cloud please? 
It would generate less logs during execution time depending on the OS

Reply:
I nested Linux & Windows tasks